docs(linter): add executor examples (#12906)
nrwl/nx: packages/linter/docs/eslint-examples.md

Linter can be configured in multiple ways. The basic way is to provide only `lintFilePatterns`, which is mandatory property. This tells us where to look for files to lint.

`project.json`:

```json
"lint": {
  "executor": "@nrwl/linter:eslint",
  "options": {
    "lintFilePatterns": ["apps/frontend/**/*.ts"]
  }
}
```

## Examples

{% tabs %}
{% tab label="Fixing linter issues" %}

> Linter provides automated way of fixing known issues. To ensure that those changes are properly cached, we need to add also `outputs`. Skipping `outputs` would produce invalid cache record. Both of these properties are set by default when scaffolding the new project.

Reviewer: ```suggestion
Linter provides an automated way of fixing known issues. To ensure that those changes are properly cached, we need to add an `outputs` property to the `lint` target. Omitting the `outputs` property would produce an invalid cache record. Both of these properties are set by default when scaffolding a new project.
```

```json
"lint": {
  "executor": "@nrwl/linter:eslint",
  "outputs": ["{options.outputFile}"],
  "options": {
    "lintFilePatterns": ["apps/frontend/**/*.ts"]
  }
}
```

> We can now run the command with `--fix` flag:

Reviewer: ```suggestion
With these settings, we can run the command with a `--fix` flag:
```

```bash
nx run frontend:lint --fix
```

> We can also set this command via project configuration to make a more permanent and not opt-in:

Reviewer: ```suggestion
We can also set this flag via project configuration to always fix files when running lint:
```
